### PR TITLE
workflows: add dispatch workflow for Linux

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

- Rename macOS workflow
- Add Linux workflow
- The end of the Linux workflof is the same as the mac one
- The Linux workflow uses the linux container for builds, so I did not not add the setup/cleanup steps

### CODE REVIEW

**Comment by iMichka** [comment]
I have a question for the test-bot experts here: if I use `brew test-bot --keep-old --only-formulae zlib` for example with this Linux workflow: what kind of recursive dependants will be tested? All of them, or only those which already have a bottle?

I would like to avoid long-running builds (especially at the beginning) and test only recursive dependants that have a bottle at the beginning and ignore the others.

Another important note: this uses the GitHub action runners. If we hit too long builds, we will need to add the same workflow yet again, but with linux self-hosted runners.

cc @jonchang @MikeMcQuaid @dawidd6

File: .github/workflows/linux-dispatch-build-bottle.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  HOMEBREW_DEVELOPER: 1
+  HOMEBREW_GITHUB_ACTIONS: 1
+  HOMEBREW_NO_AUTO_UPDATE: 1
+
+jobs:
+  bottle:
+    runs-on: ubuntu-latest
+    container:
+      image: homebrew/ubuntu16.04:master
+    env:
```
Comment (dawidd6) [inline .github/workflows/linux-dispatch-build-bottle.yml:26]: I believe those 2 HOMEBREW_GIT env vars are unneeded.

File: .github/workflows/linux-dispatch-build-bottle.yml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      HOMEBREW_GIT_NAME: BrewTestBot
+      HOMEBREW_GIT_EMAIL: 'homebrew-test-bot@lists.sfconservancy.org'
+      HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK: 1
+    steps:
+      - name: ${{github.event.inputs.formula}}
+        id: print_details
+        run: |
+          echo sender=${{github.event.sender.login}}
+          echo formula=${{github.event.inputs.formula}}
+          echo version=${{github.event.inputs.macos}}
```
Comment (dawidd6) [inline .github/workflows/linux-dispatch-build-bottle.yml:36]: This is for macOS.

File: .github/workflows/linux-dispatch-build-bottle.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    if: github.event.inputs.upload
+    env:
+      HOMEBREW_FORCE_HOMEBREW_ON_LINUX: 1
+    steps:
+      - name: ${{github.event.inputs.formula}}
+        id: print_details
+        run: |
+          echo sender=${{github.event.sender.login}}
+          echo formula=${{github.event.inputs.formula}}
+          echo version=${{github.event.inputs.macos}}
```
Comment (dawidd6) [inline .github/workflows/linux-dispatch-build-bottle.yml:107]: Same as above

**Comment by dawidd6** [comment]
We gonna need some adjusting in `brew dispatch-build-bottle` source to support linux too. I can take care of this once we have the workflow on master.

File: .github/workflows/linux-dispatch-build-bottle.yml (hunk 1 above)
Comment (iMichka) [reply .github/workflows/linux-dispatch-build-bottle.yml:26]: A lot of tests expect a git user and email to be set. Or we can set these for each test manually, but it was convenient to have these here.

**Comment by dawidd6** [comment]
Also some consistency on linux workflow files naming would be good I think. Should we go with linux prefix (tests-linux.yml) or suffix (like here)?

File: .github/workflows/linux-dispatch-build-bottle.yml (hunk 2 above)
Comment (iMichka) [reply .github/workflows/linux-dispatch-build-bottle.yml:36]: Removed.

File: .github/workflows/linux-dispatch-build-bottle.yml (hunk 3 above)
Comment (iMichka) [reply .github/workflows/linux-dispatch-build-bottle.yml:107]: Removed.

**Comment by iMichka** [comment]
I think a suffix is better, especially because the suffixes have the same length. I'll do a new pass on the other linux workflows in a second PR when I add the self-hosted runner and GitHub actions workflows.

File: .github/workflows/linux-dispatch-build-bottle.yml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+jobs:
+  bottle:
+    runs-on: ubuntu-latest
+    container:
+      image: homebrew/ubuntu16.04:master
+    env:
+      HOMEBREW_GIT_NAME: BrewTestBot
+      HOMEBREW_GIT_EMAIL: 'homebrew-test-bot@lists.sfconservancy.org'
+      HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK: 1
```
Comment (Bo98) [inline .github/workflows/linux-dispatch-build-bottle.yml:29]: This environment variable has been removed and no longer exists.

**Comment by Bo98** [comment]
I assume it's impossible to combine the two because of the container stuff?

File: .github/workflows/linux-dispatch-build-bottle.yml (hunk 4 above)
Comment (iMichka) [reply .github/workflows/linux-dispatch-build-bottle.yml:29]: It's still in the macOS dispatch build workflow though.

Comment (Bo98) [reply .github/workflows/linux-dispatch-build-bottle.yml:29]: Yeah it was removed a week or so ago and the workflows haven't been updated yet.

**Comment by iMichka** [comment]
>  assume it's impossible to combine the two because of the container stuff?

Yes. You can have a dynamic self-hosted runner workflow name, but you can't have one of the build matrix be GitHub Actions with container and the other one self-hosted runner.

File: .github/workflows/linux-dispatch-build-bottle.yml (hunk 4 above)
Comment (iMichka) [reply .github/workflows/linux-dispatch-build-bottle.yml]: I removed it from both workflows in this PR.

File: .github/workflows/linux-dispatch-build-bottle.yml (hunk 1 above)
Comment (dawidd6) [reply .github/workflows/linux-dispatch-build-bottle.yml:26]: I believe those are set in test-bot already.

Comment (dawidd6) [reply .github/workflows/linux-dispatch-build-bottle.yml:26]: If not, then Homebrew/actions/git-user-config does it fit sure

Comment (iMichka) [reply .github/workflows/linux-dispatch-build-bottle.yml]: I removed them, and we will have a look in case any build fails due to that.

**Comment by Bo98** [comment]
> what kind of recursive dependants will be tested?

No dependents are tested when running `--keep-old`.

`depends_on :linux` is an exception to this: https://github.com/Homebrew/homebrew-test-bot/blob/1e32f6ac8740808fb4fbb5c181d0b90b797bba07/lib/tests/formulae.rb#L296-L300

File: .github/workflows/macos-dispatch-build-bottle.yml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-name: Dispatch build bottle
+name: Dispatch build bottle (macOS)
```
Comment (MikeMcQuaid) [inline .github/workflows/macos-dispatch-build-bottle.yml:1]: I think the name here should stay as-is until we have merged Linux fully in here.

**Review by MikeMcQuaid — COMMENTED** [review]
Fine with this being merged as-is. Ideally we'll start pulling more of these steps into Homebrew/actions; there's starting to be a lot of duplication between them (not just these two files but across all workflows).

File: .github/workflows/macos-dispatch-build-bottle.yml (hunk 5 above)
Comment (dawidd6) [reply .github/workflows/macos-dispatch-build-bottle.yml:1]: Should we also not rename the workflow file for now?

**Comment by dawidd6** [comment]
@iMichka I let myself to force-push a reverting change regarding the macOS workflow naming. Will merge the workflow now to master as it won't break anything and later tweaks could be done on master too.

**Comment by iMichka** [comment]
Thanks!

**Comment by dawidd6** [comment]
First successful workflow after some fixes: https://github.com/Homebrew/homebrew-core/runs/2023551459?check_suite_focus=true

**Pull request merged**